tests/unit/commands/test_cmdutils.py

```diff
@@ -349,6 +349,24 @@ def fun(foo, bar, opt=False):
         with pytest.raises(IndexError):
             cmd.get_pos_arg_info(2)
 
+    def test_keyword_only_without_default(self):
+        # https://github.com/The-Compiler/qutebrowser/issues/1872
+        def fun(*, target):
+            """Blah."""
+            pass
+
+        with pytest.raises(TypeError):
+            fun = cmdutils.register()(fun)
```

Comment:
I'd prefer checking the actual exception message too here, as `TypeError` is quite broad - in other words, do `with pytest.raises(TypeError) as excinfo:` and add a `assert str(excinfo.value) == "fun: handler has keyword only argument without default!"` (or whatever the updated message will be)